includes/rest/class-outbox-controller.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * Outbox Controller file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use Activitypub\Activity\Activity;
+use Activitypub\Collection\Actors;
+use Activitypub\Collection\Outbox;
+use Activitypub\Transformer\Factory;
+use function ActivityPub\get_rest_url_by_path;
+
+/**
+ * ActivityPub Outbox Controller.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://www.w3.org/TR/activitypub/#outbox
+ */
+class Outbox_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = '(?:users|actors)/(?P<user_id>[\w\-\.]+)/outbox';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'user_id' => array(
+						'description'       => 'The ID of the user or actor.',
+						'type'              => 'string',
+						'validate_callback' => array( $this, 'validate_user_id' ),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( 'Activitypub\Rest\Server', 'verify_signature' ),
+					'args'                => array(
+						'page'     => array(
+							'description' => 'Current page of the collection.',
+							'type'        => 'integer',
+							'default'     => 1,
+							'minimum'     => 1,
+						),
+						'per_page' => array(
+							'description' => 'Maximum number of items to be returned in result set.',
+							'type'        => 'integer',
+							'default'     => 10,
+							'minimum'     => 1,
+							'maximum'     => 100,
+						),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( 'Activitypub\Rest\Server', 'verify_signature' ),
```

Comment:
posting should not be depending on the signature, but on an authentication method!